src/meson.build

```diff
@@ -33,6 +33,7 @@ lc3lib = library('lc3',
 		lc3_sources,
 		dependencies: m_dep,
 		include_directories: inc,
+		soversion: 0,
```

Comment:
A soversion of 0 implies more or less the same thing as having no soversion at all, i.e. no real ABI stability has been declared. I wonder if it would be more interesting to track ABI, instead?

Reply:
Since liblc3 is now used by [pipewire](https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/201fd7755d8dcd40284eccf65781bf38780c1f91), distros will start to package it and having a soversion (even 0) will help. Of course, it will be very good to track ABI, based on the @asoulier 's https://github.com/google/liblc3/issues/10#issuecomment-1252965472 I don't expect too much ABI instability, but I may be wrong :)

Reply:
> Since liblc3 is now used by [pipewire](https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/201fd7755d8dcd40284eccf65781bf38780c1f91), distros will start to package it and having a soversion (even 0) will help.

This is what I don't really understand. Either way, the library has a soname, but before this change, the soname was `-Wl,-soname,liblc3.so`; with this change, it is `-Wl,-soname,liblc3.so.0`.

Adding an explicit soname of 0 is explicitly saying "I confirm that there is no version", it cannot be used to schedule ABI rebuilds.

Sure, it's harmless to add. I have experience with several distros that do not care about it for good or for ill, but I suppose Debian is different. (Truthfully, I have no clue what Debian would do with it though, and confess to being slightly curious.)

> Of course, it will be very good to track ABI, based on the @asoulier 's [#10 (comment)](https://github.com/google/liblc3/issues/10#issuecomment-1252965472) I don't expect too much ABI instability, but I may be wrong :)

Agreed, this should be the preferred route if it's an option (and it does seem to be).

Reply:
Yes, I can confirm that the ABI will certainly be stable.
If it's common to Debian to add version 0, I'm fine with that.
@daissi Can you confirm this ?

Reply:
@asoulier, yes it is common to have a soversion to 0 for shared lib and it's better than no version.

Several Debian mechanisms rely on the soversion. Not having any versioning information triggers some warnings [1].
And the binary package name is based on the soversion [2].

Having this change upstream and not on the packaging side would be lovely :-)

[1] https://lintian.debian.org/tags/shared-library-lacks-version
[2] https://www.debian.org/doc/debian-policy/ch-sharedlibs.html#run-time-shared-libraries

Reply:
> Yes, I can confirm that the ABI will certainly be stable.

In that case, the ABI soversion can and should be set to `1`. :)

Reply:
Sure, I can update this MR to use `1`. @asoulier do you agree?

Reply:
Yes, following the template `name.so.major-version` sounds good to me.

Reply:
Done.

Reply:
@daissi Thanks !
@eli-schwartz Looks good to you?